ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -0,0 +1,384 @@
+import React, { useEffect } from 'react';
+import {
+  Heading,
+  Container,
+  FormControl,
+  Box,
+  FormLabel,
+  Input,
+  FormHelperText,
+  Grid,
+  Stack,
+  Button,
+  useToast,
+  Select,
+} from '@chakra-ui/react';
+import { set, useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { customAxios as axios } from '../../helpers/customAxios';
+
+const SermonNotesEditorModal = (props) => {
+  const { user, editSermonNotesData, actionOnEditor, setIsEditorOpen } = props;
+  const {
+    register,
+    handleSubmit,
+    reset,
+    setValue,
+    formState: { errors },
+  } = useForm();
+  const toast = useToast();
+
+  // Create a object
+  const sermonIdMap = {
+    'Sunday Celebration': 'sn',
+    Encounter: 'encounter',
+  };
+
+  const [title, setTitle] = useState(undefined);
+  const [sermondId, setSermonId] = useState(undefined);
+  const [subtitle, setSubtitle] = useState(undefined);
+  const [speaker, setSpeaker] = useState(undefined);
+  const [sermonSeries, setSermonSeries] = useState(undefined);
+  const [imageLink, setImageLink] = useState(undefined);
+  const [originalContent, setOriginalContent] = useState(undefined);
+  const [sermonLink, setSermonLink] = useState(undefined);
+  const [serviceType, setServiceType] = useState(undefined);
+  const [passage, setPassage] = useState(undefined);
+  const [date, setDate] = useState(undefined);
```

Comment:
im p sure all of them are string. it's better to use `''` as their default value than undefined. If u put undefined it might crash with some weird bugs in the future